doc/Architecture.md

```diff
@@ -53,29 +53,164 @@ It contains all the XAML, converters, templates, styles, assets, and other UI re
 This layer also contains platform-specific implementations of services.
 For that reason, it also contains a good portion of the configuration.
 
-## Topics
+# Technical Overview
 
-### T01 - Material Theme
+## General
+Topics that apply to the whole application.
 
-This app uses [Uno.Themes](https://github.com/unoplatform/Uno.Themes) resources.
+### Startup
+This application uses startup classes to set itself up by doing the following:
+- Select the **environment**.
+- Select the **configuration** options based on that environment.
+- **Register** the services in the dependency injection container.
+- **Start** the services and do the initial **navigation**.
 
-Resources from **Uno.Material** are used for the following:
-- Color system
-- Typography (TextBlock styles)
-- Controls styles
+This application uses 2 startup classes:
+- `Startup` is used to setup the Uno application (WinUI, Android, iOS).
+  <br/>It is invoked from App.xaml.cs, which is the entry point of the Uno application.
+- `CoreStartup` is used to setup the UI-independant services of the application.
+  <br/>It allows to have a functional app code without any UI, which is useful for testing.
+
+`CoreStartup` is used by the `Startup` class and by automated tests.
+
+See [Startup.md](Startup.md) for more details.
+
+### Dependency Injection
+This application is designed to use dependency injection (DI) to manage dependencies between components.
+
+See [DependencyInjection.md](DependencyInjection.md) for more details.
+
+### Diagnostics
+This application contains several built-in diagnostics tools. Most of them are configurable and can be can be used in release builds, which allows better support for production issues.
```

Comment:
Should it be `This application contains several built-in diagnostic tools.`. No `s` in diag?

Reply:
no s, I'll correct it.